perceval/components/source.py

```diff
@@ -181,18 +181,21 @@ def probability_distribution(self, nphotons: int = 1) -> SVDistribution:
                 svd.add(StateVector(BasicState([len(photons)], {0: photons})), prob)
         return svd
 
-    def generate_distribution(self, expected_input: BasicState):
+    def generate_distribution(self, expected_input: BasicState, simplify: bool = False):
         """
         Simulates plugging the photonic source on certain modes and turning it on.
         Computes the input probability distribution
 
         :param expected_input: Expected input BasicState
-        The properties of the source will alter the input state. A perfect source always delivers the expected state as
-        an input. Imperfect ones won't.
+            The properties of the source will alter the input state. A perfect source always delivers the expected state
+            as an input. Imperfect ones won't.
+        :param simplify: Simplify the distribution by anonymizing photon annotations (can be time-consuming for larger
+            distributions). Default is False.
         """
         dist = SVDistribution()
+        prob_threshold = global_params['min_p']
```

Comment:
What does min_p means ? minimum probability ? and why is it so low (1e-16) ?

Reply:
Yes, it's the minimum probability we consider Perceval-wide. It's low because we don't want to scrap too many "low contribution" from states. But in several algorithms (e.g. the noisy algo in Simulator), we're not using this global value as threshold.